service/models.py

```diff
@@ -0,0 +1,371 @@
+
+    @classmethod
+    def init_db(cls, app):
+        """Initializes the database session
+
+        :param app: the Flask app
+        :type data: Flask
+
+        """
+        logger.info("Initializing database")
+        cls.app = app
+        # This is where we initialize SQLAlchemy from the Flask app
+        db.init_app(app)
+        app.app_context().push()
+        db.create_all()  # make our sqlalchemy tables
+
+    @classmethod
+    def all(cls):
+        """Returns all of the Wishlists in the database"""
+        logger.info("Processing all Wishlists")
+        return cls.query.all()
+
+    @classmethod
+    def find(cls, wishlist_id):
+        """Finds a Wishlist by it's ID
+
+        :param wishlist_id: the id of the Wishlist to find
+        :type wishlist_id: int
+
+        :return: an instance with the wishlist_id, or None if not found
+        :rtype: Wishlist
+
+        """
+        logger.info("Processing lookup for id %s ...", wishlist_id)
+        return cls.query.get(wishlist_id)
+
+    @classmethod
+    def find_or_404(cls, wishlist_id):
+        """Find a Wishlist by it's id
+
+        :param wishlist_id: the id of the Wishlist to find
+        :type wishlist_id: int
+
+        :return: an instance with the wishlist_id, or 404_NOT_FOUND if not found
+        :rtype: Wishlist
+
+        """
+        logger.info("Processing lookup or 404 for id %s ...", wishlist_id)
+        return cls.query.get_or_404(wishlist_id)
+
+    @classmethod
+    def find_by_name(cls, name):
+        """Returns all Wishlists with the given name
+
+        :param name: the name of the Wishlists you want to match
+        :type name: str
+
+        :return: a collection of Wishlists with that name
+        :rtype: list
+
+        """
+        logger.info("Processing name query for %s ...", name)
+        return cls.query.filter(cls.name == name)
+
+    @classmethod
+    def find_by_customer(cls, customer):
+        """Returns all of the Wishlists in a customer
+
+        :param customer: the customer of the Wishlists you want to match
+        :type customer: int
+
+        :return: a collection of Wishlists in that customer
+        :rtype: list
+
+        """
+        logger.info("Processing customer query for %s ...", customer)
+        return cls.query.filter(cls.customer == customer)
+
+class Item(db.Model):
+    """
+    Class that represents a Item
+
+    This version uses a relational database for persistence which is hidden
+    from us by SQLAlchemy's object relational mappings (ORM)
+    """
+
+    __tablename__ = 'item'
+
+    app = None
+
+    ##################################################
+    # Table Schema
+    ##################################################
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(63), nullable=False)
+    wishlist_id = db.Column(db.Integer, db.ForeignKey('wishlist.id'))
+    wish2item = db.relationship('Wishlist', backref='wish2item')
```

Comment:
hmm, i think the relationship should be defined on `Wishlist` model... something like `relationship('Item', backref='wishlist')`, similar to the sqlalchemy example from the docs: https://docs.sqlalchemy.org/en/14/orm/backref.html#backref-arguments

as it is now one could access an `Item` via `wishlist.wish2item`, right? i think that is somewhat unexpected behavior